src/main/java/Deadline.java

```diff
@@ -0,0 +1,30 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+public class Deadline extends Task {
+    protected LocalDate by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = LocalDate.parse(by);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + this.generateByDateString() + ")";
+    }
```

Comment:
I like how you overrided the toString method to print out the appropriate message rather than having another method to print that message.